rules/sort-objects.ts

```diff
@@ -42,28 +44,29 @@ import { complete } from '../utils/complete'
 import { pairwise } from '../utils/pairwise'
 import { matches } from '../utils/matches'
 
-type Options = [
-  Partial<{
-    destructuredObjects: { groups: boolean } | boolean
-    type: 'alphabetical' | 'line-length' | 'natural'
-    customGroups: Record<string, string[] | string>
-    partitionByComment: string[] | boolean | string
-    newlinesBetween: 'ignore' | 'always' | 'never'
-    specialCharacters: 'remove' | 'trim' | 'keep'
-    locales: NonNullable<Intl.LocalesArgument>
-    groups: (Group[] | Group)[]
-    partitionByNewLine: boolean
-    objectDeclarations: boolean
-    styledComponents: boolean
-    /**
-     * @deprecated for {@link `destructuredObjects`} and {@link `objectDeclarations`}
-     */
-    destructureOnly: boolean
-    ignorePattern: string[]
-    order: 'desc' | 'asc'
-    ignoreCase: boolean
-  }>,
-]
+type Options = Partial<{
+  useConfigurationIf: {
+    allElementNamesMatchPattern?: string
```

Comment:
Maybe we should shorten it to just `pattern' or `matchPattern`?

Reply:
@azat-io I wanted to make it clear that **all** key **names** have to match for the configuration to be used (to make a difference if we decide to add a filter where **one** key has to match). I think we can simplify it to `allNamesMatchPattern`